pkg/jobmanager/manager.go

```diff
@@ -211,5 +229,15 @@ func (m *Manager) findJobByUser(userID, jobID string) (Job, error) {
 		}
 	}
 
-	return nil, fmt.Errorf("job '%s' does not exist", jobID)
+	return nil, NewJobNotFoundError(jobID)
+}
+
+// validateJobID ensures that the given jobID is in the supported format.
+// If it is not, it returns an InvalidJobID error.
+func validateJobID(jobID string) error {
```

Comment:
I wouldn't bother distinguishing "invalid" from "unknown", less code to write. Up to you.